mir-ci/mir_ci/tests/test_mir_flutter_app.py

```diff
@@ -0,0 +1,439 @@
+    Walk Pointer To ${pos}
+    Press LEFT Button
+    Walk Pointer To (600, 0)
+    Release LEFT Button
+
+Set Top Left Custom Positioner
+    Move Pointer To ${BUTTON_CUSTOM_PRESET}
+    Click And Sleep
+    Walk Pointer To ${BUTTON_SET_DEFAULTS}
+    Click And Sleep
+    # Set parent anchor to 'topLeft'
+    ${pos}=    Walk Pointer To ${DROPDOWN_PARENT_ANCHOR_LABEL}
+    Click And Sleep
+    ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 4)
+    Walk Pointer To ${pos}
+    Click And Sleep
+    # Set child anchor to 'bottomRight'
+    ${pos}=    Walk Pointer To ${DROPDOWN_CHILD_ANCHOR_LABEL}
+    Click And Sleep
+    ${pos}=    Displace ${pos} By (0, ${VERTICAL_DISTANCE_BETWEEN_OPTIONS} * 5)
+    Walk Pointer To ${pos}
+    Click And Sleep
+    Walk Pointer To ${BUTTON_APPLY}
+    Click And Sleep
+"""
+
+
+@pytest.mark.parametrize("server", servers(ServerCap.MIR_FLUTTER_APP))
+@pytest.mark.deps(
+    pip_pkgs=(
+        ("robotframework~=6.1.1", "robot"),
+        ("rpaframework", "RPA"),
+        ("rpaframework-recognition", "RPA.recognition"),
+    ),
+)
+class TestMirFlutterApp:
+    async def test_mir_flutter_app(self, robot_log, server, tmp_path) -> None:
+        extensions = VirtualPointer.required_extensions + ScreencopyTracker.required_extensions
+        server_instance = DisplayServer(server, add_extensions=extensions)
+
+        robot_test_cases = dedent(
+            """\
+            Reference App Opens
+                Sleep    2
+                VIDEO.Match    ${WINDOW_MAIN}
+
+            Regular Window Opens
+                Move Pointer To ${BUTTON_REGULAR}
+                Click And Sleep
+                VIDEO.Match    ${WINDOW_REGULAR_0_FOCUSED}
+                Close Focused Toplevel Window
+
+            Floating Regular Window Opens
+                Move Pointer To ${BUTTON_FLOATING_REGULAR}
+                Click And Sleep
+                VIDEO.Match    ${WINDOW_FLOATING_REGULAR_0_FOCUSED}
+                Close Focused Toplevel Window
+
+            Dialog Window Opens
+                Move Pointer To ${BUTTON_DIALOG}
+                Click And Sleep
+                VIDEO.Match    ${WINDOW_DIALOG_FOCUSED}
+                Close Focused Toplevel Window
+
+            Satellite Window Opens
+                Move Pointer To ${BUTTON_REGULAR}
+                Click And Sleep
+                Walk Pointer To ${BUTTON_SATELLITE_0}
+                Click And Sleep
+                Walk Pointer To ${WINDOW_SATELLITE_1}
+                Click And Sleep
+                Close Focused Toplevel Window
+
+            Popup Window Opens
+                Move Pointer To ${BUTTON_REGULAR}
+                Click And Sleep
+                Walk Pointer To ${BUTTON_POPUP_0}
+                Click And Sleep
+                Walk Pointer to ${WINDOW_POPUP_1}
+                Click And Sleep
+                Close Focused Toplevel Window
+
+            Tip Window Opens
```

Comment:
Why do some of these not have a `VIDEO.match`?

Reply:
`Walk Pointer To` internally calls `VIDEO.Match`, but it also moves the pointer. If the image isn't found, the test will fail. In general, I use `VIDEO.Match` when the pointer doesn't need to move, and `Walk Pointer To` otherwise.

Reply:
Sure, looks more expressive. I'll do it.

Reply:
Looking again at the test suite, I'm now afraid that changing `Walk Pointer To` to `Walk Pointer To And Confirm Screen State` may imply that `Move Pointer To` doesn't confirm screen state, which is false. We could change `Move Pointer To` to `Move Pointer To And Confirm Screen State` as well. However, given that both move and walk actions always "confirm screen state", maybe it would be better just to change the instances of `VIDEO.Match` to something more descriptive like `Expect ${template} Is Visible` or something similar. Would that be enough?

Reply:
Eh this is a bit of a nit in any event, so I would be ok with keeping it as-is to avoid changing a bunch of things at once. It is easily verifiable in my opinion :)